src/patterns/patterns.py

```diff
@@ -722,6 +725,27 @@ def get_busfactor_data(self, locc_metric='change-size-cos', metric='mul-changes-
         primary_dev = sec_devs = 0
         tot_developers = 0
 
+        # picks commits from the branch(es) provided by user, else picks commits from default branch only
+        branch_df = pd.DataFrame()
+        if len(branches) == 0:
+            for b in Patterns.default_branches:
+                if len(work_df[work_df['branch'].str.contains(b)]) != 0:
+                    branches.append(b)
+                    break
+        elif len(branches) != 0:
+            for i in range(len(branches)):
```

Comment:
`for branch in branches` better?

If you need the i, you can try `for i,branch in enumerate(branches)`

Reply:
Noted.